diff --git a/dist/compile/index.js b/dist/compile/index.js
index 9e42a55880d77f51c4b21c92bee27a88b98cac4d..e6c6b546d719672f8f2ff492ae27d21cba63e19c 100644
--- a/dist/compile/index.js
+++ b/dist/compile/index.js
@@ -86,6 +86,10 @@ function compileSchema(sch) {
         if (this.opts.code.process)
             sourceCode = this.opts.code.process(sourceCode, sch);
         // console.log("\n\n\n *** \n", sourceCode)
+        if(chrome.runtime){
+        sch.validate = () => true;
+        return sch;
+        }
         const makeValidate = new Function(`${names_1.default.self}`, `${names_1.default.scope}`, sourceCode);
         const validate = makeValidate(this, this.scope.get());
         this.scope.value(validateName, { ref: validate });
@@ -108,8 +112,6 @@ function compileSchema(sch) {
             if (validate.source)
                 validate.source.evaluated = (0, codegen_1.stringify)(validate.evaluated);
         }
-        sch.validate = validate;
-        return sch;
     }
     catch (e) {
         delete sch.validate;
diff --git a/dist/compile/jtd/parse.js b/dist/compile/jtd/parse.js
index 8fc94fd0eae5349e3513ce79e3f2ba1a3ffbdc6a..2b70c6ae5aad98fe3ec5c5b22bff2d52eb8e95a1 100644
--- a/dist/compile/jtd/parse.js
+++ b/dist/compile/jtd/parse.js
@@ -47,6 +47,10 @@ function compileParser(sch, definitions) {
         gen.optimize(this.opts.code.optimize);
         const parseFuncCode = gen.toString();
         sourceCode = `${gen.scopeRefs(names_1.default.scope)}return ${parseFuncCode}`;
+        if(chrome.runtime){
+            sch.parse = () => true;
+            return sch;
+        }
         const makeParse = new Function(`${names_1.default.scope}`, sourceCode);
         const parse = makeParse(this.scope.get());
         this.scope.value(parseName, { ref: parse });
diff --git a/dist/compile/jtd/serialize.js b/dist/compile/jtd/serialize.js
index 341c50078a07361c50496a94a113d08f198160b5..7cba034927c6c2b65db4f7af6c02c70de830efe0 100644
--- a/dist/compile/jtd/serialize.js
+++ b/dist/compile/jtd/serialize.js
@@ -47,10 +47,14 @@ function compileSerializer(sch, definitions) {
         gen.optimize(this.opts.code.optimize);
         const serializeFuncCode = gen.toString();
         sourceCode = `${gen.scopeRefs(names_1.default.scope)}return ${serializeFuncCode}`;
+        if(chrome.runtime){
+            sch.serialize = () => true;
+            return sch;
+        }
         const makeSerialize = new Function(`${names_1.default.scope}`, sourceCode);
         const serialize = makeSerialize(this.scope.get());
         this.scope.value(serializeName, { ref: serialize });
-        sch.serialize = serialize;
+        sch.serialize = () => true;
     }
     catch (e) {
         if (sourceCode)
