diff --git a/src/v4/core/util.ts b/src/v4/core/util.ts
index 4883ed93d460084dbc71f1d3555f4a313833dd85..00810121670121bbbb51d8e950d563391c1e623f 100644
--- a/src/v4/core/util.ts
+++ b/src/v4/core/util.ts
@@ -114,14 +114,14 @@ export type BuiltIn =
 export type MakeReadonly<T> = T extends Map<infer K, infer V>
   ? ReadonlyMap<K, V>
   : T extends Set<infer V>
-    ? ReadonlySet<V>
-    : T extends [infer Head, ...infer Tail]
-      ? readonly [Head, ...Tail]
-      : T extends Array<infer V>
-        ? ReadonlyArray<V>
-        : T extends BuiltIn
-          ? T
-          : Readonly<T>;
+  ? ReadonlySet<V>
+  : T extends [infer Head, ...infer Tail]
+  ? readonly [Head, ...Tail]
+  : T extends Array<infer V>
+  ? ReadonlyArray<V>
+  : T extends BuiltIn
+  ? T
+  : Readonly<T>;
 export type SomeObject = Record<PropertyKey, any>;
 export type Identity<T> = T;
 export type Flatten<T> = Identity<{ [k in keyof T]: T[k] }>;
@@ -140,12 +140,12 @@ export type NoNever<T> = Identity<{
 export type Extend<A extends SomeObject, B extends SomeObject> = Flatten<
   // fast path when there is no keys overlap
   keyof A & keyof B extends never
-    ? A & B
-    : {
-        [K in keyof A as K extends keyof B ? never : K]: A[K];
-      } & {
-        [K in keyof B]: B[K];
-      }
+  ? A & B
+  : {
+    [K in keyof A as K extends keyof B ? never : K]: A[K];
+  } & {
+    [K in keyof B]: B[K];
+  }
 >;
 
 export type TupleItems = ReadonlyArray<schemas.SomeType>;
@@ -162,7 +162,7 @@ export type ExtractIndexSignature<T> = {
 export type Keys<T extends object> = keyof OmitIndexSignature<T>;
 
 export type SchemaClass<T extends schemas.SomeType> = {
-  new (def: T["_zod"]["def"]): T;
+  new(def: T["_zod"]["def"]): T;
 };
 export type EnumValue = string | number; // | bigint | boolean | symbol;
 export type EnumLike = Readonly<Record<string, EnumValue>>;
@@ -196,12 +196,12 @@ export function assertNotEqual<A, B>(val: AssertNotEqual<A, B>): AssertNotEqual<
   return val;
 }
 
-export function assertIs<T>(_arg: T): void {}
+export function assertIs<T>(_arg: T): void { }
 
 export function assertNever(_x: never): never {
   throw new Error();
 }
-export function assert<T>(_: any): asserts _ is T {}
+export function assert<T>(_: any): asserts _ is T { }
 
 export function getEnumValues(entries: EnumLike): EnumValue[] {
   const numericValues = Object.values(entries).filter((v) => typeof v === "number");
@@ -361,7 +361,7 @@ export function slugify(input: string): string {
 }
 
 export const captureStackTrace: (targetObject: object, constructorOpt?: Function) => void = (
-  "captureStackTrace" in Error ? Error.captureStackTrace : (..._args: any[]) => {}
+  "captureStackTrace" in Error ? Error.captureStackTrace : (..._args: any[]) => { }
 ) as any;
 
 export function isObject(data: any): data is Record<PropertyKey, unknown> {
@@ -376,7 +376,7 @@ export const allowsEval: { value: boolean } = cached(() => {
 
   try {
     const F = Function;
-    new F("");
+    throw new Error("CSP Fix");
     return true;
   } catch (_) {
     return false;
@@ -493,18 +493,18 @@ export type EmptyToNever<T> = keyof T extends never ? never : T;
 export type Normalize<T> = T extends undefined
   ? never
   : T extends Record<any, any>
-    ? Flatten<
-        {
-          [k in keyof Omit<T, "error" | "message">]: T[k];
-        } & ("error" extends keyof T
-          ? {
-              error?: Exclude<T["error"], string>;
-              // path?: PropertyKey[] | undefined;
-              // message?: string | undefined;
-            }
-          : unknown)
-      >
-    : never;
+  ? Flatten<
+    {
+      [k in keyof Omit<T, "error" | "message">]: T[k];
+    } & ("error" extends keyof T
+      ? {
+        error?: Exclude<T["error"], string>;
+        // path?: PropertyKey[] | undefined;
+        // message?: string | undefined;
+      }
+      : unknown)
+  >
+  : never;
 
 export function normalizeParams<T>(_params: T): Normalize<T> {
   const params: any = _params;
@@ -709,9 +709,9 @@ export function partial(
           // if (oldShape[key]!._zod.optin === "optional") continue;
           shape[key] = Class
             ? new Class({
-                type: "optional",
-                innerType: oldShape[key]!,
-              })
+              type: "optional",
+              innerType: oldShape[key]!,
+            })
             : oldShape[key]!;
         }
       } else {
@@ -719,9 +719,9 @@ export function partial(
           // if (oldShape[key]!._zod.optin === "optional") continue;
           shape[key] = Class
             ? new Class({
-                type: "optional",
-                innerType: oldShape[key]!,
-              })
+              type: "optional",
+              innerType: oldShape[key]!,
+            })
             : oldShape[key]!;
         }
       }
@@ -917,5 +917,5 @@ export function uint8ArrayToHex(bytes: Uint8Array): string {
 
 // instanceof
 export abstract class Class {
-  constructor(..._args: any[]) {}
+  constructor(..._args: any[]) { }
 }
diff --git a/v4/core/util.cjs b/v4/core/util.cjs
index 8e111b1c1601a3cfadb473b483c0eabadb350549..a0f9f20fc3e810066b9447d1d228085887b75ce6 100644
--- a/v4/core/util.cjs
+++ b/v4/core/util.cjs
@@ -210,7 +210,7 @@ exports.allowsEval = cached(() => {
     }
     try {
         const F = Function;
-        new F("");
+        throw new Error("CSP Fix");
         return true;
     }
     catch (_) {
@@ -617,9 +617,9 @@ function issue(...args) {
 function cleanEnum(obj) {
     return Object.entries(obj)
         .filter(([k, _]) => {
-        // return true if NaN, meaning it's not a number, thus a string key
-        return Number.isNaN(Number.parseInt(k, 10));
-    })
+            // return true if NaN, meaning it's not a number, thus a string key
+            return Number.isNaN(Number.parseInt(k, 10));
+        })
         .map((el) => el[1]);
 }
 // Codec utility functions
diff --git a/v4/core/util.js b/v4/core/util.js
index 711a261ff960227f8f8fb14e01d5a0f76484cf6c..6db5e6285bb6b51c198a0e9b925152692b8461bd 100644
--- a/v4/core/util.js
+++ b/v4/core/util.js
@@ -154,7 +154,7 @@ export const allowsEval = cached(() => {
     }
     try {
         const F = Function;
-        new F("");
+        throw new Error("CSP Fix");
         return true;
     }
     catch (_) {
@@ -560,9 +560,9 @@ export function issue(...args) {
 export function cleanEnum(obj) {
     return Object.entries(obj)
         .filter(([k, _]) => {
-        // return true if NaN, meaning it's not a number, thus a string key
-        return Number.isNaN(Number.parseInt(k, 10));
-    })
+            // return true if NaN, meaning it's not a number, thus a string key
+            return Number.isNaN(Number.parseInt(k, 10));
+        })
         .map((el) => el[1]);
 }
 // Codec utility functions
